core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -1164,6 +1167,50 @@ public int getSigOpCount() throws ScriptException {
     }
 
     /**
+     * Generate script data bytes to represent the given block height.
+     */
+    public static byte[] generateHeightScriptData(final int height) {
+        final byte[] int32Buffer = ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN).putInt(height).array();
+
+        // Repack the buffer down to three bytes if we can
+        if (int32Buffer[3] == 0) {
+            return Arrays.copyOf(int32Buffer, 3);
+        } else {
+            return int32Buffer;
+        }
+    }
+
+    /**
+     * Check block height is in coinbase input script, for use after BIP 34
+     * enforcement is enabled.
+     */
+    public void checkCoinbaseHeight(final int height)
+            throws VerificationException {
+        // Check block height is in coinbase input script
```

Comment:
Added. Also realised it's "CoinBase" elsewhere, changing the method name to match.